src/ALE/regrid_solvers.F90

```diff
@@ -18,15 +18,19 @@ module regrid_solvers
 !! The matrix A must be square, with the first index varing down the column.
 subroutine solve_linear_system( A, R, X, N, answer_date )
   integer,              intent(in)    :: N  !< The size of the system
-  real, dimension(N,N), intent(inout) :: A  !< The matrix being inverted [nondim]
-  real, dimension(N),   intent(inout) :: R  !< system right-hand side [A]
-  real, dimension(N),   intent(inout) :: X  !< solution vector [A]
+  real, dimension(N,N), intent(inout) :: A  !< The matrix being inverted in arbitrary units [A] on
+                                            !! input, but internally modified to become nondimensional
+                                            !! during the solver.
+  real, dimension(N),   intent(inout) :: R  !< system right-hand side in arbitrary units [A B] on
+                                            !! input, but internally modified to have units of [B]
+                                            !! during the solver
+  real, dimension(N),   intent(inout) :: X  !< solution vector in arbitrary units [B]
   integer,    optional, intent(in)    :: answer_date  !< The vintage of the expressions to use
   ! Local variables
-  real, parameter       :: eps = 0.0        ! Minimum pivot magnitude allowed
-  real    :: factor       ! The factor that eliminates the leading nonzero element in a row.
-  real    :: pivot, I_pivot ! The pivot value and its reciprocal [nondim]
-  real    :: swap_a, swap_b
+  real, parameter       :: eps = 0.0        ! Minimum pivot magnitude allowed [A]
+  real    :: factor       ! The factor that eliminates the leading nonzero element in a row [A-1]
+  real    :: pivot, I_pivot ! The pivot value and its reciprocal, in [A] and [A-1]
+  real    :: swap_a, swap_b ! Swap space in various units [various]
```

Comment:
What is the difference between "various" and "[A]"?

Reply:
Variables with [various] units are used in multiple different ways that are not reliably consistent with the particular units of the rest of the variables in a routine.  By contrast, the variables in a routine with the arbitrary units of [A] all share the same units (the actual units are determined by the caller), and variables with [A-1] use the inverse of those units.

Reply:
I think I understand, but this all really needs to be documented ASAP.